### DISCUSSION TITLE
Why overriding reflected columns require type parameter?

### DISCUSSION THREAD

**Discussion by holymonson** [sqlalchemy/sqlalchemy#12163] [Usage Questions]
Test code as below. If we override reflected column without specifying `type` parameter, it will be overrided to `NullType`. IMHO, this is not necessary, we want it unchanged if not providing a new type to replace.

```python
metadata = MetaData()
metadata.reflect(engine)
metadata.tables['table_111']
# Table('table_111', MetaData(), Column('col_name', INTEGER(), table=<table_111>), schema=None)

Table(
    "table_111",
    metadata,
    Column('col_name', nullable=False),
    extend_existing=True,
)
# Table('table_111', MetaData(), Column('col_name', NullType(), table=<table_111>, nullable=False), schema=None)
```

**Comment by zzzeek** [comment]
that case was never considered as overriding types was the primary reason extend_existing is there.    For fine grained control over how a column gets set up you can use the [column_reflect](https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.DDLEvents.column_reflect) event, modify the `column_info` dictionary in place
